api/uniter/cloud_native_test.go

```diff
@@ -0,0 +1,56 @@
+package uniter_test
+
+import (
+	jc "github.com/juju/testing/checkers"
+	gc "gopkg.in/check.v1"
+	"gopkg.in/juju/environschema.v1"
+
+	"github.com/juju/juju/apiserver/facades/client/application"
+)
+
+type cloudNativeUniterSuite struct {
+	uniterSuite
+}
+
+var _ = gc.Suite(&cloudNativeUniterSuite{})
+
+func (s *cloudNativeUniterSuite) SetUpTest(c *gc.C) {
+	s.uniterSuite.SetUpTest(c)
+
+	// Ensure the application is not trusted prior to tests.
+	s.setApplicationTrust(c, false)
+}
+
+// setApplicationTrust updates the configuration for the application unit to
+// allow or deny access for cloud spec retrieval.
+func (s *cloudNativeUniterSuite) setApplicationTrust(c *gc.C, trusted bool) {
+	conf := map[string]interface{}{application.TrustConfigOptionName: trusted}
```

Comment:
While the two test below seem wonderful and warranted this test seems somewhat redundant in that we know that the application configuration feature works? `UpdateApplicationConfig` is shown to work in other tests. The purpose of building this feature on top of application config was to reuse known working/tested features. What does this test test other than the Application Configuration feature?

Reply:
This isn't a test as such; the method is called to test the trusted versus untrusted behaviour. I was thinking of extensions to this suite when I set it false by default in _SetUpTest_, but the converse might be preferable if tests are added - we would be testing trusted scenarios more than untrusted.

As for the assertions themselves, I guess the later 2 might be done away with given that the config setting is proved working by other tests, but I would definitely keep the nil error assertion for the config update; that could fail for any old downstream reason and we want to assert before running the rest of the test.

Reply:
oh, indeed, forgive me. I see that this is just a setup method.